script.js

```diff
@@ -1,94 +1,188 @@
+    this.author = author;
+  }
 }
 
-/* Using local storage */
+// display the todo in the DOM;
+class UI {
+  static displayData() {
+    const displayData = todoArr.map((item) => `
+            <div class="todo">
+            <div>
+            <span>"${item.title}" by</span>
+            <span> ${item.author}</span>
+            </div>
+            <button class="remove" data-id = ${item.id}>Remove</button>
+            </div>
+            `);
+    lists.innerHTML = (displayData).join(' ');
+  }
 
-let storedDataArray = [];
-let b_id = 0;
+  static displayNothing() {
+    const displayData = '<h3 class="display-nothing">You have no book to show</h2>';
+    lists.innerHTML = displayData;
+  }
 
-function dataStore() {
-  localStorage.setItem('name', JSON.stringify(storedDataArray));
-}
+  static colorChanger() {
+    const todos = document.querySelectorAll('.todo');
+    let count = 1;
+    todos.forEach((item) => {
+      if (count % 2 !== 0) {
+        item.classList.add('active');
+      } else if (item.classList.contains('active')) {
+        item.classList.remove('active');
+      }
+      count += 1;
+    });
+  }
 
-form.addEventListener('submit', (event) => {
-  event.preventDefault();
-  const storedData = {
-    bookID: null,
-    title: '',
-    author: '',
-  };
-  storedData.bookID = b_id;
-  storedData.title = titleValue.value;
-  storedData.author = authorValue.value;
-  storedDataArray.push(storedData);
-  dataStore();
-  add(storedData.bookID, storedData.title, storedData.author);
-  titleValue.value = null;
-  authorValue.value = null;
-  b_id += 1;
-});
+  static clearInput() {
+    bookTitle.value = '';
+    bookAuthor.value = '';
+  }
 
-if (JSON.parse(localStorage.getItem('name') !== null)) {
-  storedDataArray = JSON.parse(localStorage.getItem('name'));
-  b_id = storedDataArray[storedDataArray.length - 1].bookID + 1;
-  console.log(b_id);
-  if (storedDataArray !== null) {
-    for (let x = 0; x < storedDataArray.length; x += 1) {
-      add(storedDataArray[x].bookID, storedDataArray[x].title, storedDataArray[x].author);
+  static removeTodo() {
+    lists.addEventListener('click', (e) => {
+      if (e.target.classList.contains('remove')) {
+        e.target.parentElement.remove();
+      }
+      const btnId = e.target.dataset.id;
+      // remove from array.
+      UI.removeArrayTodo(btnId);
+    });
+  }
+
+  static removeArrayTodo(id) {
+    todoArr = todoArr.filter((item) => item.id !== +id);
+    Storage.addTodStorage(todoArr);
+    if (todoArr.length === 0) {
+      UI.displayNothing();
     }
+    UI.colorChanger();
   }
 }
+
+// form part
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const id = Math.random() * 1000000;
+  const todo = new Todo(id, bookTitle.value, bookAuthor.value);
+  todoArr = [...todoArr, todo];
+  UI.displayData();
+  UI.colorChanger();
+  UI.clearInput();
+  // add to storage
+  Storage.addTodStorage(todoArr);
+/*   if (lists.classList.contains('hide')) {
```

Comment:
Great job team 🤝

- [ ] [Optional] Always remove commented code from your project as it is regarded as a bad practice.